rfcs/engine-developer-guide.md

```diff
@@ -0,0 +1,128 @@
+
+*Recommendation*: Developers who understand where to split their content should *Split at time of submission*. For developers who are unsure, they should *Submit to developer guide* and ideally request a reviewer from sig-docs-community for evaluating content for split, to create an issue to split content when the feature is stable.
+
+**Example 2 - Gem development File locations**
+
+Bill, a project developer looking for information on custom Gem development, comes to the o3de.org site looking for documentation. Bill isn't interested in the workings
+of the Gem system; Only in the process of constructing a Gem from templates and integrating it with their project. There's now the following point of confusion: Is this information in the "User Guide", or the "Engine Developer Guide"?
+
+This RFC recommends the following to mitigate this sort of problem:
+
+1. There should still be a clear split between user-facing and developer-facing documentation. Where this is in question, the Engine Developer Guide reviewers should engage sig-docs-community's documentation reviewer pool.
+1. The Engine Developer Guide should come with a highly visible warning on the main index page, or a persistent banner, which indicates the user is in this guide.
+1. sig-docs-community should regularly evaluate user feedback and site behavior, and update any individual pages that seem to be causing problems with an appropriate notice about content in the User Guide.
+
+#### Branching
+
+The Engine Developer Manual structure and page serving information will be available only on the development branch, staged at https://development--o3deorg.netlify.app/. We recommend that the Engine Developer Manual also ships in a locked and versioned edition as part of any release.
+
+#### Information architecture
+
+The Engine Developer Manual will have page content hosted at `/docs/engine-dev-guide` (source: `/content/docs/engine-dev-guide`) and image/video materials hosted at `/images/engine-dev-guide` (source: `/static/images/engine-dev-guide`).
+
+The following broad architecture for the Engine Developer Manual is as follows (paths are relative to `https://o3de.org/docs/engine-dev-guide`):
+
+```
+_index.md           # Overview page, maintained jointly by o3de/maintainers and sig-docs-community
+architecture/*      # Broad architecture documents
+framework/*         # Per-framework documentation.
+framework/_index.md # Index linking to all framework pages, with a brief overview of the framework.
+#framework/azcore   # Naming example; Frameworks should be named after their associated binary products.
+gem/*               # Per-gem documentation. Only provided for Gems included in the O3DE distribution.
+gem/_index.md       # Index linking to all Gem pages, with a brief overview of each Gem from a technical level.
+#gem/networking     # Naming example; Gems should be named after the gem name, not any produced artifacts.
```

Comment:
Suggest we add a `tool` directory for the developer docs for included, standalone tools, such as Project Manager.
```suggestion
#gem/networking     # Naming example; Gems should be named after the gem name, not any produced artifacts.
tool/*              # Per-tool documentation for standalone tools included in the O3DE distribution.
tool/_index.md      # Index linking to all tool pages, with a brief overview of each tool.
#tool/project-manager # Naming example; Tools should be named after the tool's common program name, not necessarily the executable filename.
```